### PR TITLE
feat(pipeline): Pipeline center and edge sync action

### PR DESCRIPTION
**Pull request by kakj-go**
#### What this PR does / why we need it:
Pipeline center and edge synchronization action

#### Which issue(s) this PR fixes:

- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTc4LDExNzQsMTE5MCwxMjE4XSwiYXNzaWduZWUiOlsiMTAwMDU2MCJdfQ%3D%3D&id=307533&iterationID=1190&pId=306714&type=TASK)

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |       Pipeline center and edge synchronization action       |
| 🇨🇳 中文    |      流水线中心和边缘进行同步 action        |

### CODE REVIEW

File: modules/pipeline/providers/clusterinfo/interface.go [hunk 1]
```diff
@@ -38,7 +39,7 @@ type Interface interface {
 	// GetClusterByName return ClusterInfo query by name, contain cluster connection info and config map data
 	GetClusterInfoByName(string) (apistructs.ClusterInfo, error)
 	// ListAllClusterInfos return all clusters, contain cluster connection info and config map data
-	ListAllClusterInfos() ([]apistructs.ClusterInfo, error)
```
Comment (sfwn) [inline modules/pipeline/providers/clusterinfo/interface.go:41]: add a new method: ListEdgeClusterInfos

File: modules/pipeline/providers/edgepipeline_register/interface.go [hunk 2]
```diff
@@ -40,9 +40,18 @@ type Interface interface {
 
 	ShouldDispatchToEdge(source, clusterName string) bool
 	GetEdgeBundleByClusterName(clusterName string) (*bundle.Bundle, error)
+	ClusterIsEdge(clusterName string) bool
 	//RegisterEdgeToDialer(ctx context.Context)
 }
 
+func (p *provider) ClusterIsEdge(clusterName string) bool {
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/interface.go:47]: unclear function.
what's the difference between `IsEdge` and this?
can you use `ShouldDispatchToEdge`?

File: modules/pipeline/providers/edgepipeline_register/interface.go [hunk 3]
```diff
@@ -40,9 +40,18 @@ type Interface interface {
...
+	isEdge, err := p.bdl.IsClusterDialerClientRegistered(clusterName, apistructs.ClusterDialerClientTypePipeline.String())
+	if !isEdge || err != nil {
+		return false
```
Comment (sfwn) [inline modules/pipeline/providers/edgepipeline_register/interface.go:50]: return err

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/providers/edgepipeline_register/interface.go (hunk 2 above)
Comment (kakj-go) [reply modules/pipeline/providers/edgepipeline_register/interface.go:47]: IsEdge determines whether the current cluster is an edge cluster. The data is obtained from the configuration file.

ClusterIsEdge determines whether a cluster is an edge cluster, and the data is queried.

ShouldDispatchToEdge determines whether the source and cluster need to be synchronized to the edge. Then, if the cluster is the current cluster, it returns false

**Review by sfwn — APPROVED** [review]

**Pull request merged**